File: app/src/main/java/fr/gstraymond/autocomplete/response/AutocompleteResult.java
package fr.gstraymond.autocomplete.response;

import java.util.ArrayList;
import java.util.List;

public class AutocompleteResult {

    private Suggest suggest;

    public AutocompleteResult() {
        this.suggest = new Suggest();
        this.suggest.setCard(new ArrayList<Card>());
    }

    public void setSuggest(Suggest suggest) {
        this.suggest = suggest;
    }

    public List<Option> getResults() {
        List<Option> results = new ArrayList<>();
        for (Card card : suggest.getCard()) {
            results.addAll(card.getOptions());
        }
        return results;
    }
}

File: app/src/main/kotlin/fr/gstraymond/models/autocomplete/response/AutocompleteResult.kt
package fr.gstraymond.models.autocomplete.response

import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class AutocompleteResult(val suggest: Suggest) {
    companion object {
        fun empty() = AutocompleteResult(Suggest(listOf()))
    }

    fun getResults(): List<Option> = suggest?.card?.flatMap {it.options ?: listOf() } ?: listOf()
}

@JsonClass(generateAdapter = true)
data class Suggest(val card: List<Card>)

@JsonClass(generateAdapter = true)
data class Card(val options: List<Option>?)

@JsonClass(generateAdapter = true)
data class Option(val text: String,
                  val _source: Payload?)

@JsonClass(generateAdapter = true)
data class Payload(val stdEditionCode: String?,
                   val colors: List<String>?,
                   val type: String?,
                   val land: List<String>?)
